lib/topic_view.rb

```diff
@@ -413,9 +413,15 @@ def has_bookmarks?
   end
 
   def bookmarks
-    @bookmarks ||= @topic.bookmarks.where(user: @user).joins(:topic).select(
-      :id, :post_id, "topics.id AS topic_id", :for_topic, :reminder_at, :name, :auto_delete_preference
-    )
+    if SiteSetting.use_polymorphic_bookmarks
+      @bookmarks ||= Bookmark.for_user_in_topic(@user&.id, @topic.id).select(
```

Comment:
I don't think we need to execute a query when for an anon user. We can just return an empty array early if `@user&.id` is `nil`